### PR TITLE
Revising documentation interfaces

### PR DESCRIPTION
**Pull request by laserkelvin**
# ALCHEMI Toolkit-Ops Pull Request

## Description

This PR provides a various documentation updates in both docstrings and `sphinx` documentation.

Note that this is currently still a draft, so please do not merge; please align on them first.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

- Aligned on using `jax.Array` instead of deprecated `jnp.ndarray` for typing
- Fix various docstring inconsistencies
- Updated the sphinx user guides to use split tabs depending on which interface is being used

## Testing

- [ ] Unit tests pass locally (`make pytest`)
- [ ] Linting passes (`make lint`)
- [ ] New tests added for new functionality meets coverage expectations?

## Checklist

- [ ] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [ ] I have performed a self-review of my code
- [ ] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

**Review by dallasfoster — APPROVED** [review]

File: nvalchemiops/jax/interactions/electrostatics/ewald.py [hunk 1]
```diff
@@ -1082,13 +1082,18 @@ def ewald_reciprocal_space(
...
             total_charge_per_atom = total_charge[batch_idx]
+            volume_per_atom = volume[batch_idx]
         else:
             total_charge_per_atom = total_charge[0]
+            volume_per_atom = volume[0]
 
-        background_grad = PI / (alpha_val * alpha_val) * total_charge_per_atom
+        background_grad = (
+            PI / (2.0 * alpha_val * alpha_val * volume_per_atom) * total_charge_per_atom
+        )
```
[the remark this answers is not in the document]
Comment (dallasfoster) [reply nvalchemiops/jax/interactions/electrostatics/ewald.py:1096]: This effectively matches the warp implementation https://github.com/NVIDIA/nvalchemi-toolkit-ops/blob/640f8c6bfd28a29280916f491f5e0b3ea1ca98ca/nvalchemiops/interactions/electrostatics/ewald_kernels.py#L2013 (note that in the warp kernels we incorporate the volume in the structure factors).
